### PR TITLE
cloudtest: Run kind export kubeconfig as part of setup

### PR DESCRIPTION
**Pull request by philip-stoev**
Run `kind export kubeconfig` as part of setup to ensure that the kubeconfig file is properly populated regardless of whether the cluster existed previously or was newly-created.

### Motivation

  * This PR fixes a previously unreported bug.
Cloudtest was failing because of a missing kubeconfig.
### Tips for reviewer

@benesch this line was present before your refactoring but was removed as part of it.

Apparently for some reason, it needs to run otherwise CI fails as follows:

```
Running plugin cloudtest command hook | 0s
-- | --
  | kind: Make sure kind is running... | 8s
  | $ kubectl --context=kind-cloudtest apply -f misc/kind/configmaps/coredns.yaml
  | W0104 11:01:15.574339      52 loader.go:221] Config not found: kubeconfig
```

In other words:
1. `kind get clusters` reports a cluster exists on the given CI machine
2. still, kubeconfig is not present on that same CI machine
I have no explanation for this except that buildkite may be cleaning up the home directory between runs.

### CODE REVIEW

**Comment by philip-stoev** [comment]
Abandoned in favor of https://github.com/MaterializeInc/materialize/pull/16934

**Comment by benesch** [comment]
I thiiiink we're good here now. Might take a bit longer for everything to get rebased over #16934, but hopefully just another day or two, and we can always kill any agents that get corrupted in the meantime.
